.github/workflows/lighthouse.yml

```diff
@@ -0,0 +1,84 @@
+name: Generate Lighthouse report
+
+on:
+  issue_comment:
+    types: [edited]
+
+jobs:
+  generate_lighthouse_report:
+    timeout-minutes: 30
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout to repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+          ref: master
+
+      - name: Add Lighthouse progress comment
+        id: generate_lighthouse_comment
+        uses: marocchino/sticky-pull-request-comment@v2
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          number: ${{github.event.issue.number}}
+          header: generate_lighthouse_comment
+          message: "⏳ **Generating Lighthouse report...**"
+
+      - name: Capture Vercel preview URL
+        id: vercel_preview_url
+        uses: binary-com/vercel-preview-url-action@v1.0.5
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Generate Lighthouse report
+        uses: treosh/lighthouse-ci-action@v9
+        id: lighthouse_report
+        with:
+          temporaryPublicStorage: true
+          urls: ${{ steps.vercel_preview_url.outputs.vercel_preview_url }}
+          uploadArtifacts: true
+          # configure Lighthouse score assertions in this file
+          # configPath: .github/lighthouse/lighthouserc.json
```

Comment:
If required, we can add Lighthouse assertions to our PR checks which can be configured in the added `lighthouserc.json` with an `error` key instead of `warn`